Fix test_nans()

### Environment

* Operating System:
* Python Version: `$ python --version`
* How did you install modin-spreadsheet: (`pip`, `conda`, or `other (please explain)`)
* Python packages: `$ pip freeze` or `$ conda list` (please include modin-spreadsheet, notebook, and jupyterlab versions)
* Jupyter lab packages (if applicable): `$ jupyter labextension list`

### Description of Issue
pytest does not pass test_nans
Error from pytest:
`FAILED modin_spreadsheet/tests/test_grid.py::test_nans - TypeError: unsupported operand type(s) for +: 'int' and 'str'`
Error from traceback:
`modin_spreadsheet/grid.py:1130: TypeError`

### Reproduction Steps

1. pytest

--- a/modin_spreadsheet/grid.py
+++ b/modin_spreadsheet/grid.py
@@ -1165,7 +1165,7 @@ class SpreadsheetWidget(widgets.DOMWidget):
             )
             # Record mixed type column sort
             # Create stringified helper column to sort on
-            helper_col = self._sort_field + self._sort_col_suffix
+            helper_col = str(self._sort_field) + self._sort_col_suffix
             self._record_transformation(
                 (
                     f"{constants.SORT_MIXED_TYPE_COLUMN}\n"

--- a/modin_spreadsheet/tests/test_grid.py
+++ b/modin_spreadsheet/tests/test_grid.py
@@ -1055,3 +1055,32 @@ def test_filter_relevant_history():
     filtered_history = spreadsheet.filter_relevant_history(persist=False)
     assert filtered_history == expected_filtered_history
     assert spreadsheet.get_history() == mixed_history
+
+
+def test_mixed_type_sort():
+    df = pd.DataFrame(
+        {
+            "Mixed": [pd.Timestamp("2017-02-02"), np.nan, 1e10, 5, "hey"],
+            1: [5, 4, "hello", 2, 1],
+        }
+    )
+
+    view = SpreadsheetWidget(df=df)
+
+    view._handle_view_msg_helper(
+        {"type": "change_sort", "sort_field": "Mixed", "sort_ascending": True}
+    )
+
+    expected_order_mixed = [1e10, pd.Timestamp("2017-02-02"), 5, "hey", np.nan]
+    sorted_order_mixed = list(view.get_changed_df()["Mixed"])
+    assert expected_order_mixed[:4] == sorted_order_mixed[:4]
+    # np.nan != np.nan by definition, so check if value at index 4 are both np.nan
+    assert isnan(expected_order_mixed[4]) and isnan(sorted_order_mixed[4])
+
+    # check sorting on number column names works
+    view._handle_view_msg_helper(
+        {"type": "change_sort", "sort_field": 1, "sort_ascending": True}
+    )
+    expected_order_1 = [1, 2, 4, 5, "hello"]
+    sorted_order_1 = list(view.get_changed_df()[1])
+    assert expected_order_1 == sorted_order_1

--- a/modin_spreadsheet/tests/test_grid.py
+++ b/modin_spreadsheet/tests/test_grid.py
@@ -1065,22 +1065,22 @@ def test_mixed_type_sort():
         }
     )
 
-    view = SpreadsheetWidget(df=df)
+    spreadsheet = SpreadsheetWidget(df=df)
 
-    view._handle_view_msg_helper(
+    spreadsheet._handle_view_msg_helper(
         {"type": "change_sort", "sort_field": "Mixed", "sort_ascending": True}
     )
 
     expected_order_mixed = [1e10, pd.Timestamp("2017-02-02"), 5, "hey", np.nan]
-    sorted_order_mixed = list(view.get_changed_df()["Mixed"])
+    sorted_order_mixed = list(spreadsheet.get_changed_df()["Mixed"])
     assert expected_order_mixed[:4] == sorted_order_mixed[:4]
     # np.nan != np.nan by definition, so check if value at index 4 are both np.nan
-    assert isnan(expected_order_mixed[4]) and isnan(sorted_order_mixed[4])
+    assert np.isnan(expected_order_mixed[4]) and np.isnan(sorted_order_mixed[4])
 
     # check sorting on number column names works
-    view._handle_view_msg_helper(
+    spreadsheet._handle_view_msg_helper(
         {"type": "change_sort", "sort_field": 1, "sort_ascending": True}
     )
     expected_order_1 = [1, 2, 4, 5, "hello"]
-    sorted_order_1 = list(view.get_changed_df()[1])
+    sorted_order_1 = list(spreadsheet.get_changed_df()[1])
     assert expected_order_1 == sorted_order_1

